### PR TITLE
Clarify Helm chart release instructions.

### PR DESCRIPTION
**Pull request by charleskorn**
#### What this PR does

Clarifies the release steps for the Helm chart.

#### Which issue(s) this PR fixes or relates to

(none)

#### Checklist

- [n/a] Tests updated
- [x] Documentation added
- [n/a] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: operations/helm/charts/mimir-distributed/RELEASE.md [hunk 1]
```diff
@@ -9,32 +9,45 @@ Either:
...
    1. Update the `home` URL to point to the appropriate documentation version in the Helm [Chart.yaml](https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/Chart.yaml). This should be the closest Mimir documentation version, not "latest" or "next" that change content.
    1. Run `make doc`, to update [README.md](https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/README.md) from its template
-   1. Open a PR
+   1. Open a PR.
+
+      For stable versions, ensure your PR targets the `mimir-distributed-release-x.y` branch you created above.
+
+      For all other versions, your PR should target `main`.
+
    1. Merge the PR after review
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/RELEASE.md:49]:
```suggestion
   1. Merge the PR after review
   1. For stable versions, once the release tag in Git is created, merge the branch back to `main`.
```

File: operations/helm/charts/mimir-distributed/RELEASE.md [hunk 2]
```diff
@@ -9,32 +9,45 @@ Either:
...
-1. Determine version number
+1. Determine version number.
 
-   - Helm [mandates semantic versioning](https://helm.sh/docs/topics/charts/#the-chartyaml-file)
-   - Weekly release should have the version: x.y.z-weekly.w (e.g. 3.1.0-weekly.196)
-   - Release candidate: x.y.z-rc.w (e.g. 3.1.0.rc.1)
-   - Final version: x.y.z (e.g. 3.1.0)
+   Helm [mandates semantic versioning](https://helm.sh/docs/topics/charts/#the-chartyaml-file).
+
+   - Weekly releases should have the version: `x.y.z-weekly.w` (e.g. `3.1.0-weekly.196`)
```
[the remark this answers is not in the document]
Comment (charleskorn) [reply operations/helm/charts/mimir-distributed/RELEASE.md:16]: What does semi-automatically mean here? The PR is generated automatically but needs to be manually merged?

[the remark this answers is not in the document]
Comment (krajorama) [reply operations/helm/charts/mimir-distributed/RELEASE.md:16]: yes, exactly

[the remark this answers is not in the document]
Comment (charleskorn) [reply operations/helm/charts/mimir-distributed/RELEASE.md]: Fixed in 92d8001cc9c13dacedd4faf05d169ad4c31dc194.

**Review by krajorama — APPROVED** [review]

**Pull request merged**